# Cálculo - Tarefa 02

1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  
2- Calcule essa área de forma analítica.

Cálculo numérico da integral de 
𝑥
3
x 
3
  usando soma de Riemann
Para calcular a integral numericamente, dividimos o intervalo 
[
1
,
2
]
[1,2] em subintervalos e usamos retângulos para aproximar a área sob a curva 
𝑓(𝑥)=𝑥3
f(x)=x 3.

Configuração:
𝑓
(
𝑥
)
=
𝑥
3
f(x)=x 
3
 
Intervalo: 
[
1
,
2
]
[1,2]
𝑛
n: Número de retângulos (vamos usar 
𝑛
=
100
n=100 para boa precisão).
Fórmula:
A soma de Riemann é dada por:

Soma de Riemann
=
∑
𝑖
=
1
𝑛
𝑓
(
𝑥
𝑖
∗
)
Δ
𝑥
Soma de Riemann= 
i=1
∑
n
​
 f(x 
i
∗
​
 )Δx
Onde:

Δ
𝑥
=
𝑏
−
𝑎
𝑛
Δx= 
n
b−a
​
 : largura de cada retângulo.
𝑥
𝑖
∗
=
𝑎
+
𝑖
⋅
Δ
𝑥
x 
i
∗
​
 =a+i⋅Δx: ponto dentro do subintervalo usado para a altura (vamos usar o ponto médio).
Vamos calcular isso numericamente.

In [14]:
import numpy as np

# Função
f = lambda x: x**3

# Intervalo e número de retângulos
a, b = 1, 2
n = 100  # Número de retângulos
dx = (b - a) / n  # Largura dos retângulos

# Pontos médios
x_mid = np.linspace(a + dx / 2, b - dx / 2, n)

# Soma de Riemann
area_numerica = np.sum(f(x_mid) * dx)
area_numerica

3.7499624999999996

<span style="color:red;">Resultado:
A área numérica é aproximadamente 3.7500 usando 100 retângulos.</span>



In [3]:
from sympy import symbols, integrate

# Define a variável simbólica
x = symbols('x')

# Define a função
f = x**3

# Calcula a integral definida
area_analitica = integrate(f, (x, 1, 2))

# Resultado
area_analitica

15/4

In [5]:
area_decimal = area_analitica.evalf()
area_decimal

3.75000000000000

<span style="color:red;">Comparação dos resultados:  
Numérico: 
3.7500
3.7500 com 
𝑛
=
100
n=100 retângulos.  
Analítico: 
3.75
3.75 (exato).  
Ambos os métodos concordam, mostrando que a aproximação numérica é precisa com 
𝑛
=
100
n=100.</span>